### PR TITLE
Modify generate tests method

### PR DESCRIPTION
**Pull request by kimberlythegeek**
Modified the way that tests are generated in order to enable the configuration of xfail within classes, as opposed to individual `conftest.py` files.

### CODE REVIEW

File: pytest_axe/tests/test_axe.py [hunk 1]
```diff
@@ -12,29 +13,40 @@
...
+
+    def test_run(self, base_url, test_page):
+        """Assert that run method returns a non-empty dictionary."""
+        violations = test_page.run()
+        assert violations is not None
+
+    def test_parameterize_tests(self, rule, test_page):
+        """
+            Test that parameterized tests xfail correctly, based on xfail_rules
+            defined in conftest.py.
```
Comment (davehunt) [inline pytest_axe/tests/test_axe.py]: It looks like the docstring needs updating here.

**Review by davehunt — APPROVED** [review]

File: pytest_axe/parameterize_tests.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ def pytest_configure(config):
     # Make a list containing only rule IDs.
     for rule in rules:
         list.append(rule["ruleId"])
-    xfail_rules = metafunc.config.xfail_rules
+    # xfail_rules = metafunc.config.xfail_rules
```
Comment (m8ttyB) [inline pytest_axe/parameterize_tests.py]: Did you intend to remove this or leave it as commented out?

File: pytest_axe/parameterize_tests.py [hunk 3]
```diff
@@ -27,7 +27,7 @@ def axe_core_get_rules():
     return rules
 
 
-def parametrized_accessibility_rules(metafunc):
+def parametrized_accessibility_rules(xfail_rules):
```
Comment (m8ttyB) [inline pytest_axe/parameterize_tests.py]: Woot \o/

File: pytest_axe/pytest_axe.py [hunk 4]
```diff
@@ -53,9 +53,7 @@ def run_axe(page, context=None, options=None, impact=None):
 
 
 class PytestAxe(Axe):
-    def __init__(
-        self, selenium, script_url=None, context=None, options=None, impact=None
-    ):
+    def __init__(self, selenium, context=None, options=None, impact=None):
```
Comment (m8ttyB) [inline pytest_axe/pytest_axe.py:51]: Just a nit, how do you feel about adding a docstring here?

**Review by m8ttyB — CHANGES_REQUESTED** [review]
Looks great, just few items that were noted by Dave and I.

**Pull request merged**